explainaboard/feature.py

```diff
@@ -333,54 +336,61 @@ def __post_init__(self):
-                    value = str(value).strip()
-                output.append(self._str2int[str(value)])
+    def str2int_scalar(self, value: str) -> int:
+        """Convert a class name into the corresponding ID."""
+        try:
+            if self._str2int is not None:
+                return self._str2int.get(value) or self._str2int[value.strip()]
             else:
                 # No names provided, try to integerize
-                failed_parse = False
-                try:
-                    output.append(int(value))
-                except ValueError:
-                    failed_parse = True
-                if failed_parse or not 0 <= value < self.num_classes:
-                    raise ValueError("Invalid string class label %s" % value)
-        return output if return_list else output[0]
-
-    def int2str(self, values: Union[int, Iterable]):
-        """Conversion integer => class name string."""
-        assert isinstance(values, int) or isinstance(
-            values, Iterable
-        ), f"Values {values} should be an integer or an Iterable (list, numpy array, pytorch, tensorflow tensors)"
-        return_list = True
-        if isinstance(values, int):
-            values = [values]
-            return_list = False
+                output = int(value)
+                if 0 <= output < unwrap(self.num_classes):
+                    raise ValueError(f'{value} is out of range.')
+                return output
+        except Exception as ex:
+            raise ValueError(f'Invalid string class label: {value}.') from ex
+
+    def str2int_vector(self, values: Iterable[str]) -> list[int]:
+        """Convert a list of class names into corresponding IDs."""
+        return [self.str2int_scalar(v) for v in values]
+
+    def str2int(self, values: Iterable[str]) -> Union[int, list[int]]:
+        """Convert class names into corresponding IDs: generic version."""
+        # NOTE(odashi): str is also Iterable[str].
+        if isinstance(values, str):
+            return self.str2int_scalar(values)
+        else:
+            return self.str2int_vector(values)
+
+    def int2str_scalar(self, value: int) -> str:
+        """Convert a class ID into the corresponding name."""
+        if not 0 <= value < unwrap(self.num_classes):
+            raise ValueError(f'{value} is out of range.')
+
+        if self._int2str is not None:
+            return self._int2str[value]
+        else:
+            # No names provided, assuming the string representation is the name.
+            return str(value)
+
+    def int2str_vector(self, values: Iterable[int]) -> list[str]:
+        """Convert a list of class IDs into corresponding names."""
+        return [self.int2str_scalar(v) for v in values]
+
+    @overload
+    def int2str(self, values: int) -> str:
+        ...
 
-        for v in values:
-            if not 0 <= v < self.num_classes:
-                raise ValueError("Invalid integer class label %d" % v)
+    @overload
```

Comment:
Maybe I'm not familiar with this syntax, but just to confirm: is `int2str` intentionally defined twice or is this a typo?

Reply:
OK, thanks for the clarification.